test/it/api_auth_it.js

```diff
@@ -596,6 +600,70 @@ describe('Application.authenticateApiRequest',function(){
       assert.equal(result[1].tokenResponse.scope,givenScope.join(' '));
     });
 
+    describe('that is given a callback', function () {
+      it('should return an error if given one', function(done) {
+        var expectedError = new Error('expected_error');
+
+        app.authenticateApiRequest({
+          request: requestObject,
+          scopeFactory: function(account, requestedScope, callback){
+            callback(expectedError);
+          }
+        },function(err){
+          assert.equal(err, expectedError);
+          done();
+        });
+      });
+
+      it('should return an error if called multiple times', function(done) {
+        var mochaErrorListener = process.listeners('uncaughtException').pop();
+
+        process.removeListener('uncaughtException', mochaErrorListener);
+
+        process.nextTick(function () {
+          process.listeners('uncaughtException').push(mochaErrorListener);
+        });
+
+        process.once('uncaughtException', function (err) {
+          assert.equal(err.message, 'Callback has already been called once. Assert that your scopeFactory doesn\'t return a result while also calling the callback.');
+          done();
+        });
+
+        app.authenticateApiRequest({
+          request: requestObject,
+          scopeFactory: function(account, requestedScope, callback){
+            callback(null, '123');
```

Comment:
Haha yeah, though it doesn't really make any sense since that value isn't really used for anything. But I know you like your UUIDs.